frontend/ui/elements/menu_keyboard_layout.lua

```diff
@@ -43,6 +42,50 @@ local function getActivatedKeyboards(compact)
     return table.concat(activated_keyboards, ", "), #activated_keyboards
 end
 
+local function isKeyboardLayoutActive(lang)
+    if VirtualKeyboard:getKeyboardLayout() == lang then
+        return true
+    end
+
+    local keyboard_layouts = G_reader_settings:readSetting("keyboard_layouts", {})
+    return util.arrayContains(keyboard_layouts, lang)
+end
+
+-- Generate the language specific settings menu on demand, and only for active layouts,
+-- to avoid loading potentially complex/large data when the layout isn't enabled.
+local function genLayoutSpecificSubmenu()
+    local item_table = {}
+
+    for lang, __ in FFIUtil.orderedPairs(VirtualKeyboard.lang_has_submenu) do
+        if isKeyboardLayoutActive(lang) then
```

Comment:
Since this is now guarded by this check, I wonder if we actually need `lang_has_submenu` (though I guess this is technically a slight optimisation over loading every enabled keyboard as opposed to just those with submenus -- which is currently just Japanese)?

Reply:
Yeah, since the whole goal of this PR is to *not* load the keyboards, I definitely like this ;).